src/IntervalFilterAllWithTags.cpp

```diff
@@ -27,18 +27,59 @@
 #include <IntervalFilterAllWithTags.h>
 #include <Interval.h>
 
-IntervalFilterAllWithTags::IntervalFilterAllWithTags (std::set <std::string> tags): _tags (std::move(tags))
+IntervalFilterAllWithTags::IntervalFilterAllWithTags (std::set <std::string> tags, const bool complexFiltering): _tags (std::move(tags)), _complexFiltering (complexFiltering)
 {}
 
 bool IntervalFilterAllWithTags::accepts (const Interval& interval)
 {
-  for (auto& tag : _tags)
-  {
-    if (! interval.hasTag (tag))
+  if (_complexFiltering) {
```

Comment:
As you have already said, this should not be part of `IntervalFilterAllWithTags`, but rather some function or factory which translates the individual command line arguments into filters or filter groups, like e.g. `IntervalFilterOrGroup`, `IntervalFilterWithoutTagSet`, etc.